### PR TITLE
feat: concurrently load & parse fonts

### PR DESCRIPTION
**Pull request by Itsusinn**
This will reduce load time from 2376ms to 388ms when loading many huge fonts

### CODE REVIEW

**Comment by Itsusinn** [comment]
@jackpot51 Could you please take a look?

**Comment by Itsusinn** [comment]
https://docs.rs/rayon-core/1.12.1/rayon_core/index.html#global-fallback-when-threading-is-unsupported

I just find rayon supports `no_std`, maybe it's unnecessary to add `rayon` feature-gate?

**Comment by CryZe** [comment]
That's not the same as `no_std`. If the crate doesn't have a `std` feature, then it doesn't support `no_std`.

**Comment by Itsusinn** [comment]
> That's not the same as `no_std`. If the crate doesn't have a `std` feature (and otherwise uses `std` features), then it doesn't support `no_std`.

Understood

File: Cargo.toml [hunk 1]
```diff
@@ -28,14 +28,15 @@ ttf-parser = { version = "0.20.0", default-features = false }
...
+rayon = { version = "1", optional = true }
 
 [dependencies.unicode-bidi]
 version = "0.3.13"
 default-features = false
 features = ["hardcoded-data"]
 
 [features]
-default = ["std", "swash", "fontconfig"]
+default = ["std", "swash", "fontconfig", "rayon"]
```
Comment (genusistimelord) [inline Cargo.toml:39]: I would either Place Rayon under the Std feature, OR not add it to defaults and add the std as its own requirement.

Personally, adding it under std seems to be a decent choice here if you want it defaulted.

**Review by genusistimelord — CHANGES_REQUESTED** [review]
Everything But the feature itself, being added as a default, looks good.  Once that is fixed I think this would be ready to merge as I see no issues coming from this.  @jackpot51

File: Cargo.toml (hunk 1 above)
Comment (jackpot51) [reply Cargo.toml:39]: I agree

**Review by genusistimelord — APPROVED** [review]
looks much better thank you

**Review by jackpot51 — APPROVED** [review]

**Pull request merged**